torch/testing/_internal/common_utils.py

```diff
@@ -2502,8 +2485,15 @@ def make_fullrank_matrices_with_distinct_singular_values(*shape, device, dtype,
         # TODO: improve the handling of complex tensors here
         real_dtype = t.real.dtype if t.dtype.is_complex else t.dtype
         k = min(shape[-1], shape[-2])
-        s = torch.arange(1., k + 1, dtype=real_dtype, device=device).mul_(1.0 / (k + 1))
-        x = (u * s.to(dtype)) @ vh
+        # We choose the singular values to be "around one"
+        # This is to make the matrix well conditioned
+        # s = [2, 3, ..., k+1]
+        s = torch.arange(2, k + 2, dtype=real_dtype, device=device)
+        # s = [2, -3, 4, ..., (-1)^k k+1]
+        s[1::2] *= -1.
+        # 1 + 1/s so that the singular values are in the range [2/3, 3/2]
+        s.reciprocal_().add_(1.)
+        x = (u * s.to(u.dtype)) @ vh
```

Comment:
nit: maybe we should add a comment explaining that s not being sorted is not an issue here?